libpivot/sources/engine.cxx

```diff
@@ -50,17 +53,35 @@ Engine::Engine()
     m_system_index.registerSystem(builtins::systems::collisionTestSystem);
     m_system_index.registerSystem(builtins::systems::testTickSystem);
 
+    auto current_entry = boost::dll::program_location().parent_path();
+    auto find_asset_folder = [](const boost::filesystem::path &entry) -> std::optional<std::filesystem::path> {
+        for(auto& directory_entry : boost::make_iterator_range(boost::filesystem::directory_iterator(entry), {})) {
+            if (!boost::filesystem::is_directory(directory_entry))
+                continue;
+            if (directory_entry.path().filename() == "assets")
+                return std::filesystem::path(directory_entry.path().string());
+        }
+        return {};
+    };
+    for (int i = 0; i < 4; i++) {
```

Comment:
Magic number !
Replace the `4` by a constant either in code or even better in Cmake

```cpp
#ifndef PIVOT_ASSET_SEARCH_DEPTH
    #define PIVOT_ASSET_SEARCH_DEPTH 4
#endif
```
And add a private compile definition to the cmake